docs/02-guides/customize-wysiwyg-platform.mdx

```diff
@@ -0,0 +1,200 @@
+have some specific media shortcodes while Magento will have some widgets to
+display a category name. In the following section you will learn about the one
+implemented in Front-Commerce:
+
+<div className="pl-4">
+  <TOCInline toc={toc} maxHeadingLevel={4} />
+</div>
+
+Definitions:
+
+- a _shortcode_ is a specific string structure that is meant to be transformed
+  into actual content
+- a _transform_ is function that replaces an HTML tag with a React Component
+
+## `DefaultWysiwyg`
+
+The goal of this Wysiwyg type is to remain as simple as possible. It is not
+meant to fetch data from additional services.
+
+**No shortcodes**
+
+**Default transforms**
+
+- `<a>` tags are transformed into `theme/components/atoms/Typography/Link`
+  components when the `href` attribute does not contain a domain.
+
+## `MagentoWysiwyg`
+
+The MagentoWyswiyg's goal is to support all the default features in Magento 1
+& 2. If you notice that some features are missing, please <ContactLink /> and
+we'll look into implementing it.
+
+**Supported shortcodes**
+
+- `{% raw %}{{media url="*"}}{% endraw %}`
+- `{% raw %}{{store url="*"}}{% endraw %}`
+- `{% raw %}{{widget type="*" attribute="value"}}{% endraw %}`
+
+**Default transforms**
+
+- `<a>` tags are transformed into `theme/components/atoms/Typography/Link`
+  components when the `href` attribute does not contain a domain.
+- `<widget>` tags are transformed into
+  `theme/modules/Wysiwyg/MagentoWysiwyg/Widget/Widget.js` components. However,
+  you shouldn't write a `<widget>` tag manually. It comes from the
+  `{% raw %}{{widget}}{% endraw %}` shortcode.
+- `<style>` tags are transformed to support
+  [Magento's Page Builder](/docs/3.x/extensions/magento2/how-to/use-and-extend-the-page-builder)
+  format. The `#html-body` selector is replaced with the `<Wysiwyg>` root
+  selector. See
+  [WYSIWYG dynamic styles](/docs/3.x/guides/display-wysiwyg-content#dynamic-styles)
+  for details.
+
+### Add a custom Magento Widget
+
+Custom widgets will be automatically parsed. However, you will still need to map
+the widget's `type` to custom React Components. If you don't, they will be
+ignored and nothing will be rendered in the final page.
+
+To do so, please override
+`theme/modules/Wysiwyg/MagentoWysiwyg/Widget/getWidgetComponent.js` in your own
+theme.
+
+Now, within your newly created `getWidgetComponent.js` file, you will be able to
+add your own behavior. For instance, if you've created a widget in Magento that
+uses the type `acme/product-preview`, you will need to update the `Widget.js`
+file likewise:
+
+```js title="theme/modules/Wysiwyg/MagentoWysiwyg/Widget/getWidgetComponent.js"
+import React from "react";
+import Spinner from "theme/components/atoms/Spinner";
+import loadable from "@loadable/component";
+
+const defaultWidgetsMap = {
+  "acme/product-preview": loadable(
+    () => import("./path/to/ProductPreview.js"),
+    {
+      fallback: <Spinner />,
+    }
+  ),
+};
```

Comment:
We no longer support loadable, this should be replaced with react `lazy()` and `Suspense`